snooty/rstparser.py

```diff
@@ -485,6 +486,61 @@ def run(self) -> List[docutils.nodes.Node]:
         return [node]
 
 
+class TocTreeDirective(docutils.parsers.rst.Directive):
+    """Special handling for toctree directives.
+
+    Produces a node that includes an `entries` property, represented as a list of objects. Each entry in entries includes:
+    - slug OR url: a string representing the absolute url or path of the page to navigate to
+    - [optional] title: a string representing the title to use in the TOC sidebar
+    """
+
+    required_arguments = 0
+    optional_arguments = 0
+    has_content = True
+    final_argument_whitespace = True
+    option_spec = {
+        "caption": str,
+        "hidden": util.option_flag,
+        "includehidden": util.option_flag,
+        "maxdepth": int,
+        "titlesonly": util.option_flag,
+        "reversed": util.option_flag,
+        "numbered": util.option_flag,
+        "glob": util.option_flag,
+    }
+
+    def run(self) -> List[docutils.nodes.Node]:
+        source, line = self.state_machine.get_source_and_line(self.lineno)
+        node = directive(self.name)
+        node.document = self.state.document
+        node.source, node.line = source, line
+        node["options"] = self.options
+
+        entries: List[Dict[str, str]] = []
+        for child in self.content:
+            entries.append(self.make_toc_entry(source, child))
+        node["entries"] = entries
+
+        return [node]
+
+    def make_toc_entry(self, source: str, child: str) -> Dict[str, str]:
+        """Parse entry for either url or slug and optional title"""
+        entry: Dict[str, str] = {}
+        if "<" in child:
```

Comment:
Does `PAT_EXPLICIT_TILE` (um, apparently I made a typo that should be fixed) do what you need here?